### PR TITLE
Add environment to all library methods

### PR DESCRIPTION
**Pull request by cmfcmf**

### CODE REVIEW

File: src/node_lib.h [hunk 1]
```diff
@@ -178,6 +166,21 @@ NODE_EXTERN void StopEventLoop();
...
+ *
+ * Loads and executes the content of the given file.
+ * This method returns after the script was evaluated once.
+ * This means, that any pending events will not be processed as long as
+ * `ProcessEvents` or `RunEventLoop` is not called.
+ * @param path The path to the JavaScript file.
+ * @return The return value of the given JavaScript file.
+ */
+NODE_EXTERN v8::MaybeLocal<v8::Value> Run(const std::string& path);
+
```
Comment (Johannes-Schneider) [inline src/node_lib.h:182]: Please add documentation.

**Review by Johannes-Schneider — CHANGES_REQUESTED** [review]

**Comment by Johannes-Schneider** [comment]
Would it be a good idea to check, whether we can use `Initialize` to create mutliple instances of `Environment`? This would be very convenient for users who might want to use multiple `Environment`, but dont intend to write all the setup code themselves.

**Comment by cmfcmf** [comment]
> Would it be a good idea to check, whether we can use Initialize to create mutliple instances of Environment? This would be very convenient for users who might want to use multiple Environment, but dont intend to write all the setup code themselves.

I don't think we can easily change our Initialize function to accomplish this, because it initializes a lot more than just the environment (`atexit`, `PlatformInit()`, ...). I think it is out of scope for us to make it easier to create multiple environments.

File: src/node_lib.h [hunk 2]
```diff
@@ -352,34 +435,21 @@ NODE_EXTERN v8::MaybeLocal<v8::Value> Call(
...
-    v8::Local<v8::Object> receiver,
-    v8::Local<v8::Function> function,
-    std::initializer_list<v8::Local<v8::Value>> args);
-
 NODE_EXTERN v8::MaybeLocal<v8::Value> Call(
     Environment* env,
     v8::Local<v8::Object> receiver,
     v8::Local<v8::Function> function,
-    std::initializer_list<v8::Local<v8::Value>> args);
-}  // namespace node
```
Comment (cmfcmf) [inline src/node_lib.h]: What happened here?

Comment (luminosuslight) [reply src/node_lib.h]: Sorry, that was an accident, fixed it :+1:

Comment (cmfcmf) [reply src/node_lib.h]: I was just working on the same thing when you started pushing :-D I've now also pushed my commit, which should make this ready.

**Review by luminosuslight — CHANGES_REQUESTED** [review]
The env parameter has to be added to eventLoopIsRunning, start/stopEventLoop and tickEventLoop, too, and the documentation for internal::isolate()/environment() should be adapted.

I will stop working on it now so that we don't do it twice again :-D :see_no_evil:

**Comment by cmfcmf** [comment]
> The env parameter has to be added to eventLoopIsRunning, start/stopEventLoop and tickEventLoop, too, 

I'm not sure that's true. Calling tickEventLoop with a different environment meanse calling it with an environment not created by `Initialize()`. I don't think that's really needed at the moment. 

> and the documentation for internal::isolate()/environment() should be adapted.

Yes, that's right.

**Comment by luminosuslight** [comment]
I'm not sure about this, but shouldn't it be possible to call startEventLoop(env) with an environment created in the same way as in Initialize()? If not, than this parameter is not required, right. We should then document that it is using the env from the Initialize() function.

**Comment by cmfcmf** [comment]
> I'm not sure about this, but shouldn't it be possible to call startEventLoop(env) with an environment created in the same way as in Initialize()? If not, than this parameter is not required, right. We should then document that it is using the env from the Initialize() function.

Well, I think it would be possible in theory, but not worth the hassle at the moment. If you are that advanced to create own environments without using the Initialize function, you might as well write your own event loop.

**Comment by cmfcmf** [comment]
Merging based on feedback received on Telegram 🔭

**Pull request merged**